elooplib/archive.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#  define SET_BINARY_MODE(file)
+
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include "zlib.h"
+
+static constexpr size_t chunk = 16384;
+int compress(FILE *source, FILE *dest, int level)
```

Comment:
Functions inside headers should be either inline or static. But that's not the case, please use cpp file